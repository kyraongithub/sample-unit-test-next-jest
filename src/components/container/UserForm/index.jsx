import React, { useState } from 'react';
import Input from '@/components/ui/Input';
import Typography from '@/components/ui/Typography';
import Button from '@/components/ui/Button';

const UserForm = () => {
  const [name, setname] = useState('');
  const [email, setemail] = useState('');

  const onUserSubmit = () => {
    console.log('User submitted');
  };

  return (
    <div>
      <Typography type="h1">Add a user</Typography>
      <Input
        value={name}
        onChange={(e) => setname(e.target.value)}
        label="Name"
        type="text"
        placeholder="Enter name"
      />
      <br />
      <br />
      <Input
        value={email}
        onChange={(e) => setemail(e.target.value)}
        label="Email"
        type="text"
        placeholder="Enter email"
      />
      <br />
      <br />
      <Button onClick={onUserSubmit}>Submit</Button>
    </div>
  );
};

export default UserForm;
