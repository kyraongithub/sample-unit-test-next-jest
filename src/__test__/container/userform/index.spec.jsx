import UserForm from '@/components/container/UserForm';
import { fireEvent, render } from '@testing-library/react';

describe('UserForm', () => {
  it('should render user form', () => {
    const { container } = render(<UserForm />);
    expect(container).toMatchSnapshot();
  });

  it('should handle change user name', () => {
    const { getByLabelText } = render(<UserForm />);
    const input = getByLabelText('Name');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(input).toHaveValue();
  });

  it('should handle change user email', () => {
    const { getByLabelText } = render(<UserForm />);
    const input = getByLabelText('Email');
    fireEvent.change(input, { target: { value: 'test@email.com' } });
    expect(input).toHaveValue();
  });

  it('should handle submit user form', () => {
    const { container, getByText } = render(<UserForm />);
    const submit = getByText('Submit');
    fireEvent.click(submit);
    expect(container).toMatchSnapshot();
  });

  it('should handle full flow submit user form', () => {
    const { container, getByLabelText, getByText } = render(<UserForm />);
    const input = getByLabelText('Name');
    const email = getByLabelText('Email');
    const submit = getByText('Submit');
    fireEvent.change(input, { target: { value: 'test' } });
    fireEvent.change(email, { target: { value: 'test@email.com' } });
    fireEvent.click(submit);
    expect(container).toMatchSnapshot();
  });
});
