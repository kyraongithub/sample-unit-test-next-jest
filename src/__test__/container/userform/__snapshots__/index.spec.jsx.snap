// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should handle full flow submit user form 1`] = `
<div>
  <div>
    <h1>
      Add a user
    </h1>
    <label
      for="Name"
    >
      Name
    </label>
    <br />
    <input
      id="Name"
      placeholder="Enter name"
      type="text"
      value="test"
    />
    <br />
    <br />
    <label
      for="Email"
    >
      Email
    </label>
    <br />
    <input
      id="Email"
      placeholder="Enter email"
      type="text"
      value="test@email.com"
    />
    <br />
    <br />
    <button>
      Submit
    </button>
  </div>
</div>
`;

exports[`UserForm should handle submit user form 1`] = `
<div>
  <div>
    <h1>
      Add a user
    </h1>
    <label
      for="Name"
    >
      Name
    </label>
    <br />
    <input
      id="Name"
      placeholder="Enter name"
      type="text"
      value=""
    />
    <br />
    <br />
    <label
      for="Email"
    >
      Email
    </label>
    <br />
    <input
      id="Email"
      placeholder="Enter email"
      type="text"
      value=""
    />
    <br />
    <br />
    <button>
      Submit
    </button>
  </div>
</div>
`;

exports[`UserForm should render user form 1`] = `
<div>
  <div>
    <h1>
      Add a user
    </h1>
    <label
      for="Name"
    >
      Name
    </label>
    <br />
    <input
      id="Name"
      placeholder="Enter name"
      type="text"
      value=""
    />
    <br />
    <br />
    <label
      for="Email"
    >
      Email
    </label>
    <br />
    <input
      id="Email"
      placeholder="Enter email"
      type="text"
      value=""
    />
    <br />
    <br />
    <button>
      Submit
    </button>
  </div>
</div>
`;
